File: 013)OOPProject/app/src/main/java/com/example/oopproject/MainActivity.java
package com.example.oopproject;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

       /* User myuser = new User();
        myuser.name = "Elyesa";
        myuser.job = "Software Engineeiring";*/
        User myUser = new User("Elyesa", "Software Engineeiring");
        System.out.println(myUser.information());
        System.out.println(myUser.name);

        //Encapsulation
        Family elyesa = new Family("Elyesa", 25);
        System.out.println(elyesa.getName());
        elyesa.setAge(26, "Elyesa");
        System.out.println(elyesa.getAge());

        //Inheritance
        SuperFamily fatma = new SuperFamily("Fatma", 25);
        System.out.println(fatma.sing());
        System.out.println(fatma.getAge());

        //Polymorphism
        //Statik Polymorphism
        Mathematics mathematics = new Mathematics();
        System.out.println(mathematics.sum());
        System.out.println(mathematics.sum(5, 3));
        System.out.println(mathematics.sum(5, 3, 4));

        //Dynamic Polymorphism
        Animal myAnimal = new Animal();
        myAnimal.sing();

        Dog barley = new Dog();
        barley.test();
        barley.sing();

        Piano myPiano = new Piano();
        myPiano.brand = "Yamaha";
        myPiano.digital = true;
        myPiano.info();

    }
}

File: 054)KotlinOOPProject/app/src/main/java/com/elyesasimsek/kotlinoopproject/MainActivity.kt
package com.elyesasimsek.kotlinoopproject

import android.os.Bundle
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        //constructor
        var myUser = User("Elyesa",50)
        println(myUser.age.toString())
        println(myUser.name)
        println(myUser.information())

        //encapsulation
        var elyesa = Family("Elyesa", 25)
        elyesa.setAge(25)
        println(elyesa.getAge().toString())

        //inheritance
        var fatma = SuperFamily("Fatma", 25)
        println(fatma.getName())
        fatma.sing()

        //polymorohism
        //static polymorphism
        var mathematics = Mathematics()
        println(mathematics.sum())
        println(mathematics.sum(5, 6))
        println(mathematics.sum(5, 6, 7))

        //dinamic polymoophism
        val animal = Animal()
        animal.sing()

        val barley = Dog()
        barley.test()
        barley.sing()

        //abstarct & interface
        //var myPeople = People() olmuyor
        var myPiano = Piano()
        myPiano.brand = "Yamaha"
        myPiano.digital = false
        println(myPiano.roomName)
        myPiano.info()

        //lambda expressions
        fun printString(myString: String){
            println(myString)
        }

        printString("My Test String")

        val testString ={myString: String -> println(myString)}

        testString("My Lambda String")

        val multiplyLambda = {a: Int, b: Int -> a * b}
        println(multiplyLambda(5, 4))

        val mutiplyLambda2 : (Int, Int) -> Int = {a, b -> a*b}
        println(mutiplyLambda2(5, 5))

        //asynchrnous
        //callback function
        fun downloadFamily(url: String, completion:(Family) -> Unit){
            val meryem = Family("Meryem", 0)
            completion(meryem)
        }

        downloadFamily("metallica.com", { family ->
            println(family.getName())
        })











    }
}